### PR TITLE
br, tidb: Add backup and restore size for TiDB SQL brie task

### LINKED ISSUE #27247: BACKUP SQL Command returns size of 0
## Bug Report

### 1. Minimal reproduce step (Required)
#### (1) start a TiDB cluster locally through tiup 

1. env: CentOS
2. command: tiup playground
3. version: v5.1.1

#### (2) generate some data before backup
I used the following script to generate around 5 million records
```bash
#!/bin/bash

mysql -h127.0.0.1 -P4000 -uroot -e "CREATE TABLE if not exists test.user(name VARCHAR(200), age int) COMMENT='user info';"

for((i=1;i<=10000;i++));  
do   
        mysql -h127.0.0.1 -P4000 -uroot -e "INSERT INTO test.user VALUES('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30), ('AAA', 10), ('BBB', 20), ('CCC', 30);"
done  
```
#### (3) use SQL command to backup
```SQL
BACKUP DATABASE * TO 'local:///tmp/backup0816/full';
```

### 2. What did you expect to see? (Required)
After the BACKUP is finished, it should return info about the BACKUP whose size is bigger than 0.

### 3. What did you see instead (Required)
The backup took 9.04 seconds and the "Size" is 0, while I could see the backup file in folder /tmp/backup0816/full and I could use these files to retore to a new constructed empty TiDB cluster using SQL command **RESTORE DATABASE * FROM 'local:///tmp/backup0816/full';**
```SQL
+------------------------------+------+--------------------+---------------------+---------------------+
| Destination                  | Size | BackupTS           | Queue Time          | Execution Time      |
+------------------------------+------+--------------------+---------------------+---------------------+
| local:///tmp/backup0816/full |    0 | 427058788672733186 | 2021-08-16 15:46:33 | 2021-08-16 15:46:33 |
+------------------------------+------+--------------------+---------------------+---------------------+
1 row in set (9.04 sec)
```

### 4. What is your TiDB version? (Required)

```SQL
MySQL [test]> SELECT tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                       |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v5.1.1
Edition: Community
Git Commit Hash: 797bddd25310ed42f0791c8eccb78be8cce2f502
Git Branch: heads/refs/tags/v5.1.1
UTC Build Time: 2021-07-28 11:00:26
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?

Issue Number: close #27247 

Problem Summary:
SQL backup and restore show a 0 size
### What is changed and how it works?
SQL backup and restore shall show a correct size, not 0.
Proposal: [xxx](url) 

What's Changed:
add backup size
How it Works:
glue is communitation class between tidb and br, glue does not record a size for general tidb sql.
### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Please add a release note, or a 'None' if it is not needed.
```

### CODE REVIEW

**Comment by fengou1** [comment]
run-check_title

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -277,6 +278,8 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		TTL:      utils.DefaultBRGCSafePointTTL,
 		ID:       utils.MakeSafePointID(),
 	}
+	g.Record("BackupTS", restoreTS)
```
Comment (YuJuncen) [inline br/pkg/task/restore.go:281]: I guess `backupMeta.Version` would be more suitable here?

Also cc @3pointer

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

File: br/pkg/task/restore.go (hunk 1 above)
Comment (fengou1) [reply br/pkg/task/restore.go:282]: Juncen, luancheng and me have talked about this, restore and backup as one atom task in tidb, if split brie into 2 tasks, it take too much effort, big impact in tidb, so we would like to keep this change as it.

Comment (YuJuncen) [reply br/pkg/task/restore.go:282]: OK

**Review by YuJuncen — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**